src/librustc_resolve/lib.rs

```diff
@@ -2403,7 +2403,7 @@ impl<'a> Resolver<'a> {
                             (format!("maybe a missing crate `{}`?", ident), None)
                         }
                     } else if i == 0 {
-                        (format!("use of undeclared type or module `{}`", ident), None)
+                        (format!("use of undeclared crate, module, or type `{}`", ident), None)
```

Comment:
```suggestion
                        (format!("use of undeclared crate, module or type `{}`", ident), None)
```

Reply:
Grepping for ", or " in Rust sources finds that it's a commonly used form. There's an error with "not a struct, enum, or union".